stl/inc/memory

```diff
@@ -1596,18 +1887,310 @@ private:
+    if constexpr ((rank_v<_Ty>) > 0) { // TRANSITION, LLVM-42404
+        for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+            _Reverse_destroy_multidimensional_n_al_guard _Guard{_Out, _Idx, _Al};
+            _Uninitialized_copy_multidimensional_al(_Val, _Out[_Idx], _Al);
+            _Guard._Dismiss();
+        }
+    } else if constexpr (_Fill_memset_is_safe<_Ty*, _Ty> && _Uses_default_construct<_Alloc, _Ty*, _Ty>::value) {
+        (void) _Al;
+        _CSTD memset(_Out, static_cast<unsigned char>(_Val), _Size);
+    } else {
+        _Uninitialized_rev_destroying_backout_al _Backout{_Out, _Al};
+        for (size_t _Count = 0; _Count < _Size; ++_Count) {
+            _Backout._Emplace_back(_Val);
+        }
+        _Backout._Release();
+    }
+}
+
+// CLASS TEMPLATE _Ref_count_unbounded_array_alloc
+template <class _Ty, class _Alloc>
+class __declspec(empty_bases) _Ref_count_unbounded_array_alloc : public _Ebco_base<_Alloc>, public _Ref_count_base {
+    // handle reference counting for unbounded array in control block, allocator
+public:
+    explicit _Ref_count_unbounded_array_alloc(const _Alloc& _Al_arg, const size_t _Count)
+        : _Ebco_base<_Alloc>(_Al_arg), _Ref_count_base(), _Size(_Count) {
+        _Maybe_rebind_alloc_t<_Alloc, remove_cv_t<remove_all_extents_t<_Ty>>> _Alty(this->_Get_val());
+        _Uninitialized_value_construct_multidimensional_n_al(_Get_ptr(), _Size, _Alty);
+    }
+
+    explicit _Ref_count_unbounded_array_alloc(
+        const _Alloc& _Al_arg, const size_t _Count, const remove_extent_t<_Ty>& _Val)
+        : _Ebco_base<_Alloc>(_Al_arg), _Ref_count_base(), _Size(_Count) {
+        _Maybe_rebind_alloc_t<_Alloc, remove_cv_t<remove_all_extents_t<_Ty>>> _Alty(this->_Get_val());
+        _Uninitialized_fill_multidimensional_n_al(_Get_ptr(), _Size, _Val, _Alty);
+    }
+
+    auto _Get_ptr() {
+        return reinterpret_cast<remove_extent_t<_Ty>*>(
+            reinterpret_cast<char*>(this) + _Div_ceil(sizeof(*this), alignof(_Ty)) * alignof(_Ty));
+    }
+
+private:
+    size_t _Size;
+
+    ~_Ref_count_unbounded_array_alloc() = default; // nothing to do; _Storage._Value already destroyed by _Destroy()
+
+    virtual void _Destroy() noexcept override { // destroy managed resource
+        using _Rebound = _Maybe_rebind_alloc_t<_Alloc, remove_all_extents_t<_Ty>>;
+        if constexpr (!conjunction_v<is_trivially_destructible<remove_extent_t<_Ty>>,
+                          _Uses_default_destroy<_Rebound, remove_all_extents_t<_Ty>*>>) {
+            _Rebound _Alty(this->_Get_val());
+            _Reverse_destroy_multidimensional_n_al(_Get_ptr(), _Size, _Alty);
+        }
+    }
+
+    virtual void _Delete_this() noexcept override { // destroy self
+        constexpr size_t _Align = (_STD max)(alignof(_Ty), alignof(_Ref_count_unbounded_array_alloc));
+        using _Storage          = aligned_storage_t<_Align, _Align>;

```

Comment:
Billy suggested that instead of `aligned_storage_t<_Align, _Align>` we should use a `struct` containing an `alignas` `unsigned char` array. I'll work on this.

Reply:
Will push a change for this soon.